src/util/prolog/prologResult.ts

```diff
@@ -9,22 +9,32 @@ const booleanAnswerRegExp = /^(true|false)( ;|,|.)$/;
 // Second group contains the variable names
 const variableNameRegExp = /(^| )([A-Z]\w*)/g;
 
+/** Handles the results for a prolog query returned by PrologSession.executeQuery() */
 export class PrologResult {
 	private readonly rawResults: string[];
 	private results: Map<string, Array<string | boolean>> | undefined;
 
+	/**
+		* Simply forwards the raw results to its respective attribute
+		* @param {Array<string>} rawResults - The string array containing the results
+		*/
 	constructor(rawResults: string[]) {
 		this.rawResults = rawResults;
 	}
 
-	public getRawResults() {
+	/**
+		* rawResults getter-Methode
+		* @return The raw results
+		*/
+	public getRawResults(): readonly string[] {
```

Comment:
You could use `public readonly rawResults: ReadonlyArray<string>` as type instead of using an ordinary array. Then a) this getter method would be superfluous and b) you'd also enforce that no private method will ever modify these values either.